tools/database/src/analyze_contract_sizes.rs

```diff
@@ -0,0 +1,132 @@
+        ContractSizeStats {
+            topn,
+            top_accounts: BTreeMap::new(),
+            total_accounts: 0,
+            shard_accounts: BTreeMap::new(),
+        }
+    }
+
+    pub fn add_info(&mut self, shard_uid: ShardUId, account_id: AccountId, contract_size: usize) {
+        self.total_accounts += 1;
+        *self.shard_accounts.entry(shard_uid).or_insert(0) += 1;
+
+        self.top_accounts.insert(ByteSize::b(contract_size as u64), account_id);
+        if self.top_accounts.len() > self.topn {
+            self.top_accounts.pop_first();
+        }
+    }
+
+    pub fn print_stats(&self, compressed: bool) {
+        println!("");
+        println!("Analyzed {} accounts", self.total_accounts);
+        println!("Accounts per shard:");
+        for (shard_uid, count) in self.shard_accounts.iter() {
+            println!("{}: {}", shard_uid, count);
+        }
+        println!("");
+        if compressed {
+            println!("Top {} accounts by compressed contract size:", self.topn);
+        } else {
+            println!("Top {} accounts by contract size:", self.topn);
+        }
+        for (size, account_id) in self.top_accounts.iter().rev() {
+            println!("{}: {}", size, account_id);
+        }
+    }
+}
+
+impl AnalyzeContractSizesCommand {
+    pub(crate) fn run(&self, home: &PathBuf) -> anyhow::Result<()> {
+        let mut near_config = load_config(home, GenesisValidationMode::Full).unwrap();
+        let node_storage = open_storage(&home, &mut near_config).unwrap();
+        let store = node_storage.get_split_store().unwrap_or_else(|| node_storage.get_hot_store());
+        let chain_store = Rc::new(ChainStore::new(
+            store.clone(),
+            near_config.genesis.config.genesis_height,
+            false,
+        ));
+
+        let head = chain_store.head().unwrap();
+        let epoch_manager =
+            EpochManager::new_from_genesis_config(store.clone(), &near_config.genesis.config)
+                .unwrap();
+        let shard_layout = epoch_manager.get_shard_layout(&head.epoch_id).unwrap();
+        let last_block = chain_store.get_block(&head.last_block_hash).unwrap();
+        if last_block.header().chunk_mask().contains(&false) {
+            panic!("Last block doesn't have all chunks, please try again later!");
+        }
+
+        let mut stats = ContractSizeStats::new(self.topn);
+        for chunk in last_block.chunks().iter() {
+            let shard_id = chunk.shard_id();
+            let shard_uid = ShardUId::from_shard_id_and_layout(shard_id, &shard_layout);
+            println!("Analyzing chunk with uid: {}", shard_uid);
+
+            let state_root = chunk.prev_state_root();
+            let trie_storage = Rc::new(TrieDBStorage::new(store.clone(), shard_uid));
+            let trie = Trie::new(trie_storage, state_root, None);
+
+            let mut iterator = trie.iter().unwrap();
+            iterator.seek_prefix(&[col::CONTRACT_CODE]).unwrap();
```

Comment:
Alternatively we can look at flat storage head and read all data from `FlatState`. But if this is quick already / we already have all the results, proceed with Trie.

Reply:
It takes under 30s to analyze the current mainnet state on `dev-release`, so I think it's fast enough.